embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/api/ApiClient.kt

```diff
@@ -1,179 +1,25 @@
 package io.embrace.android.embracesdk.comms.api
 
-import io.embrace.android.embracesdk.logging.InternalEmbraceLogger
-import io.embrace.android.embracesdk.logging.InternalStaticEmbraceLogger
-import java.io.ByteArrayOutputStream
-import java.io.IOException
-import java.io.InputStream
-import java.io.InputStreamReader
-import java.net.HttpURLConnection
-import java.util.zip.GZIPOutputStream
+internal interface ApiClient {
+    /**
+     * Executes [ApiRequest] as a GET, returning the response as a [ApiResponse]
+     */
+    fun executeGet(request: ApiRequest): ApiResponse<String>
 
-/**
- * Client for calling the Embrace API. This service handles all calls to the Embrace API.
- *
- * Sessions can be sent to either the production or development endpoint. The development
- * endpoint shows sessions on the 'integration testing' screen within the dashboard, whereas
- * the production endpoint sends sessions to 'recent sessions'.
- *
- * The development endpoint is only used if the build is a debug build, and if integration
- * testing is enabled when calling [Embrace.start()].
- */
-internal class ApiClient @JvmOverloads constructor(
-    private val logger: InternalEmbraceLogger = InternalStaticEmbraceLogger.logger
-) {
+    /**
+     * Executes [ApiRequest] as a POST with the given body defined by [payloadToCompress], returning the response as a [ApiResponse].
+     * The body will be gzip compressed.
+     */
+    fun executePost(request: ApiRequest, payloadToCompress: ByteArray): ApiResponse<String>
```

Comment:
The one slightly significant change is to return the full response abstraction rather than just the body. This enables consumers of this components to deal with the response better (e.g. non-200 responses), which will be done in a later PR.